### PR TITLE
chore(infra): node 10+, webpack@5 compat, latest webpack-defaults

### PR DESCRIPTION
**Pull request by AviVahl**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**
- [x] infra upgrade

### Motivation / Use-Case

Getting this loader repository up-to-speed with the rest of them.

### Breaking Changes
Node 10+, as per latest webpack-defaults.

### Additional Info
- fixes webpack5 compatibility by avoiding `process.nextTick()` (which no longer gets polyfilled by default).
- drop special `MochaPhantomJS` initialization.
- avoid using `document.write()`, and instead create the output element right before running (after `DOMContentLoaded` event).
- upgrade all deps to latest stable.
- regenerate `package-lock.json`.
- closes #82

### CODE REVIEW

**Comment by evilebottnawi** [comment]
Good job, need fix for webpack@5, canary tests should not fail

**Comment by AviVahl** [comment]
will do. working on updating the template a bit more.

**Comment by AviVahl** [comment]
I'll re-open once I have a proper behavioural test in place.

The current test grabbed one of the modules and snapshot it. This approach breaks when snapshoting the entire bundle (due to differences of webpack 5).

I plan to implement a test that actually loads the bundle with `puppeteer` and verifies the output is working (meaning, bundled test is running in browser).

**Comment by evilebottnawi** [comment]
@AviVahl feel free to feedback and questions

**Comment by AviVahl** [comment]
@evilebottnawi It's all green now. care to review? :)
I've re-factored the test into a proper e2e.
I also removed the initialization code for phantomjs. Seems like a good time to drop support for it.

File: package.json [hunk 1]
```diff
@@ -37,41 +41,42 @@
   ],
   "peerDependencies": {
     "mocha": "^2.0.0 || ^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0",
```
Comment (evilebottnawi) [inline package.json]: Maybe we can drop some old `mocha` versions from peerDeps?

Comment (AviVahl) [reply package.json]: done. left 5 + 6.

**Comment by AviVahl** [comment]
Yes, thank you. Wanna do the honors and also release a major?

**Comment by evilebottnawi** [comment]
@AviVahl yes, let's do it

**Pull request merged**